video_xblock/static/js/player_state.js

```diff
@@ -25,10 +25,30 @@ var player_state = {
   'playbackRate': {{ player_state.playback_rate }},
   'muted': {{ player_state.muted | yesno:"true,false" }},
   'transcriptsEnabled': {{ player_state.transcripts_enabled | yesno:"true,false" }},
-  'captionsEnabled': {{ player_state.captions_enabled | yesno:"true,false" }}
+  'captionsEnabled': {{ player_state.captions_enabled | yesno:"true,false" }},
+  'captionsLanguage': '{{ player_state.captions_language }}',
 };
 
 var xblockUsageId = window.location.hash.slice(1);
+var transcripts = {
+  {% for transcript in player_state.transcripts %}
+   '{{transcript.lang}}': {
+          'label': '{{transcript.label}}',
+          'url': '{{transcript.url}}',
+   },
+  {% endfor %}
+};
+
+/** Get transcript url for current caption language */
+var getDownloadTranscriptUrl = function (player){
```

Comment:
After changing the captions language we need to save state on backend and  in the same time to change the download transcript url to appropriate language without reloading the page. So by that reason I've made this function here